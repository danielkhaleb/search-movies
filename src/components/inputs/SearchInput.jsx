import React, {Component} from 'react';

import InputMask from 'react-input-mask';

class SearchInput extends Component {

    render(){
        return (
            <div className="field factor-form-search-input-container">
                <label className="label">{this.props.label}</label>
                <p className="control has-icons-right">
                    <InputMask
                        className={`input ${this.props.additionalClassName || 'is-full-width'}`}
                        type="text"
                        name={this.props.name}
                        placeholder={this.props.placeholder}
                        disabled={this.props.disabled}
                        defaultValue={this.props.initialValue}
                        value={this.props.value}
                        onChange={this.props.onChange}
                        onBlur={this.props.onBlur}
                        mask={this.props.mask}
                        onKeyPress={this.props.onKeyPress}
                    />
                    <span className="icon is-small is-right">
                        <i className="fa fa-search"></i>
                    </span>
                </p>
            </div>
        );
    }
}
export default SearchInput;