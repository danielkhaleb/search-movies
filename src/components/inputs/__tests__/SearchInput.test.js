import React from 'react';
import SearchInput from '../SearchInput';
import renderer from 'react-test-renderer';

describe('Search input component', () => {

    it('renders correctly', () => {
        const tree = renderer.create(
            <SearchInput
                label="Search"
                placeholder="Title from movie"
            />
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });

});