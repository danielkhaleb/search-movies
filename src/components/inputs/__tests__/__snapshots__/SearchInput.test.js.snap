// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search input component renders correctly 1`] = `
<div
  className="field factor-form-search-input-container"
>
  <label
    className="label"
  >
    Search
  </label>
  <p
    className="control has-icons-right"
  >
    <input
      className="input is-full-width"
      onBlur={[Function]}
      onFocus={[Function]}
      placeholder="Title from movie"
      type="text"
    />
    <span
      className="icon is-small is-right"
    >
      <i
        className="fa fa-search"
      />
    </span>
  </p>
</div>
`;
